scripts/git_changes_utils.py

```diff
@@ -0,0 +1,304 @@
+    if err:
+        raise ValueError(err)
+
+    return merge_base.decode('utf-8').strip()
+
+
+def compare_to_remote(
+    remote: str, local_branch: str, remote_branch: Optional[str] = None
+) -> List[FileDiff]:
+    """Compare local with remote branch with git diff.
+
+    Parameter:
+        remote: str. Name of the git remote being pushed to.
+        local_branch: str. Name of the git branch being pushed to.
+        remote_branch: str|None. The name of the branch on the remote
+            to test against. If None, the remote branch is considered
+            to be the same as the local branch.
+
+    Returns:
+        list(FileDiff). List of FileDiffs that are modified, changed,
+        renamed or added but not deleted.
+
+    Raises:
+        ValueError. Raise ValueError if git command fails.
+    """
+    remote_branch = remote_branch if remote_branch else local_branch
+    git_remote = '%s/%s' % (remote, remote_branch)
+    # Ensure that references to the remote branches exist on the local machine.
+    common.start_subprocess_for_result(['git', 'pull', remote])
+    # Only compare differences to the merge base of the local and remote
+    # branches (what GitHub shows in the files tab of pull requests).
+    return git_diff_name_status(
+        get_merge_base(git_remote, local_branch), local_branch)
+
+
+def get_parent_branch_name_for_diff() -> str:
+    """Returns remote branch name against which the diff has to be checked.
+
+    Returns:
+        str. The name of the remote branch.
+    """
+    if common.is_current_branch_a_hotfix_branch():
+        return 'release-%s' % common.get_current_release_version_number(
+            common.get_current_branch_name())
+    return 'develop'
+
+
+def extract_acmrt_files_from_diff(diff_files: List[FileDiff]) -> List[bytes]:
+    """Grab only files out of a list of FileDiffs that have a ACMRT status."""
+    if not diff_files:
+        return []
+    acmrt_files = [f.name for f in diff_files if f.status in b'ACMRT']
+    return acmrt_files
+
+
+def get_refs() -> List[GitRef]:
+    """Returns the ref list taken from STDIN or the current branch."""
+    ref_list = []
+    if not sys.stdin.isatty():
+        # Git provides refs in STDIN.
+        ref_list = [GitRef(*ref_str.split()) for ref_str in sys.stdin]
+    if len(ref_list) == 0:
+        current_branch = common.get_current_branch_name()
+        encoded_stdout, encoded_stderr = common.start_subprocess_for_result(
+            ['git', 'show-ref', current_branch])
+        stderr = encoded_stderr.decode('utf-8')
+        if stderr:
+            raise ValueError(stderr)
+        stdout = encoded_stdout.decode('utf-8')
+        local_ref_line, remote_ref_line = stdout.splitlines()
+        local_sha, local_ref = local_ref_line.split()
+        remote_sha, remote_ref = remote_ref_line.split()
+        ref_list.append(GitRef(local_ref, local_sha, remote_ref, remote_sha))
+    if len(ref_list) > 0:
```

Comment:
```suggestion
    if ref_list != []:
```

I think this makes your intent more explicit

Reply:
Done.